src/main/kotlin/inc/evil/d17map/exceptions/GlobalExceptionHandler.kt

```diff
@@ -58,18 +54,25 @@ class GlobalExceptionHandler {
         return ResponseEntity(ErrorResponse(ex.message ?: "Invalid equipment data"), HttpStatus.BAD_REQUEST)
     }
 
-    @ExceptionHandler(AccessDeniedException::class)
-    fun handleAccessDenied(ex: AccessDeniedException): ResponseEntity<ErrorResponse> {
-        return ResponseEntity(ErrorResponse("Access denied: ${ex.message}"), HttpStatus.FORBIDDEN)
+    @ExceptionHandler(UserNotFoundException::class)
+    fun handleUserNotFound(ex: UserNotFoundException): ResponseEntity<ErrorResponse> {
+        return ResponseEntity(ErrorResponse(ex.message ?: "User not found"), HttpStatus.NOT_FOUND)
     }
 
-    @ExceptionHandler(AuthenticationException::class)
-    fun handleAuthenticationException(ex: AuthenticationException): ResponseEntity<ErrorResponse> {
-        return ResponseEntity(ErrorResponse("Authentication required: ${ex.message}"), HttpStatus.UNAUTHORIZED)
+    @ExceptionHandler(UserAlreadyExistsException::class)
+    fun handleUserAlreadyExistsException(ex: UserAlreadyExistsException): ResponseEntity<ErrorResponse> {
+        return ResponseEntity(ErrorResponse(ex.message ?: "User already exists"), HttpStatus.CONFLICT)
     }
 
-    @ExceptionHandler(UserNotFoundException::class)
-    fun handleUserNotFound(ex: UserNotFoundException): ResponseEntity<ErrorResponse> {
-        return ResponseEntity(ErrorResponse(ex.message ?: "User not found"), HttpStatus.NOT_FOUND)
+    @ExceptionHandler(RuntimeException::class)
+    fun handleGenericException(ex: RuntimeException): ResponseEntity<ErrorResponse> {
+        ex.printStackTrace()
+        if (ex is AuthenticationException || ex is AccessDeniedException)
+            throw ex
+
```

Comment:
Może zamiast `ex.printStackTrace()` lepiej byłoby użyć tutaj jakiegoś loggera?

Reply:
Tak, to na pewno, poprawię to 😊